### PR TITLE
MultiServer: Rename server_password to admin_password

### PR DESCRIPTION
**Pull request by recklesscoder**
## The Problem
I know it, you know it, everybody knows it: Having two different server settings named `password` and `server_password` is stupid. Better get this over with rather sooner than later.

## What is this fixing or adding?
This renames the `server_password` option to `admin_password` in all the places. Forwards/backwards compatibility fallbacks for everything are provided and annotated with TODOs for eventual removal, so that nothing should break right now.

"Everything" includes server_options in `host.yaml`, `.archipelago` files, and `.apsave` files, as well as the command-line option. Deprecation warnings are emitted when the old command-line option or old `host.yaml` server_options are used.

## How was this tested?
I essentially tried out each of the fallbacks, including old `.archipelago`+`.apsave` with new server code and new `.archipelago`+`.apsave` with old server code.

## Inconveniences
Every single user that uses Generate or MultiServer will get to see the warning until they manually update their `host.yaml` (or until the fallbacks are removed, at which point `server_password` would become an unknown and ignored setting). I guess it would help if the installer could edit the file to update that line. How do new games get added to the `host.yaml` files of existing installs? Could that mechanism be applied here as well?

## To go even further beyond
While we're at it, we could also rename `password` to `connect_password`, if we wanted to be really explicit about it. In my opinion, that is not necessary if `server_password` is renamed.

### CODE REVIEW

**Comment by Jarno458** [comment]
So a `password` and `server_password` is bad, but a `password` and `admin_password` is not?

I dont see how this rename add much value tbh

**Comment by recklesscoder** [comment]
I guess I was being too flippant to make my case very well. Let me try again.

The terms "password" and "server password" currently both describe server options, but looking at their names you couldn't tell that they are in fact different and separate features, nor what the difference would be. This can easily lead to confusion and mistakes, both for newcomers who have not read about these features and for experienced people who need to muster additional mental strain to not slip up and confuse the two in a moment of inattention. Additionally, anytime someone refers to one of the two features in discussion, extra effort may need to be expended to avoid or clear up misunderstandings. In short, these names are too similar, not least because the features they describe in fact both pertain to a server.

To improve the situation, I propose prefixing the password that is presumably rarer and less end-user facing with an indication of its unique aspect, leaving the more common feature to take the spot of the default "password":
- The current "password" is used to authenticate oneself as someone who may connect to a server. If I wanted to identify this password unambiguously, I would call it the "connect password" or "the password for connecting". I believe most users to come across this feature very early, namely when reading instructions on how to connect with their first game.
- The current "server password" is used to authenticate oneself as someone with administrative privileges after connecting to a server, hence the proposed name "admin password". I believe this feature to be less commonly encountered.

**Comment by ThePhar** [comment]
Since there's 2 passwords, it may be better to just be more specific for both.

`room_password` the password you join with.
`admin_password` the password to run admin commands on the server.

Thoughts?

**Comment by ThePhar** [comment]
Probably backwards compatible breaking though, so that may not be a great solution.

**Comment by CaitSith2** [comment]
If anything, the rename from `server_password` to `admin_password` is the more important one.  I wish I would have thought of using `admin_password` back when I wrote the feature to start with.

**Comment by Alchav** [comment]
> Probably backwards compatible breaking though, so that may not be a great solution.

Should be able to just check for the old setting names if they're present

**Comment by t3hf1gm3nt** [comment]
> Since there's 2 passwords, it may be better to just be more specific for both.
> 
> `room_password` the password you join with. `admin_password` the password to run admin commands on the server.
> 
> Thoughts?

i mean if we're changing it anyway, i'd much rather both password option names be descriptive. even if it does break backwards compat

**Comment by recklesscoder** [comment]
We could easily give the password for connecting the same treatment, i.e., change it and have perfect compatibility for some arbitrary transition period (could be many months), such that nothing breaks until after the transition period. Even after that period the only things that break are Generate/MultiServer executables that have not been updated the entire time and `.archipelago` and `.apsave` files that were created before the transition period (these two breakages can in fact be scheduled separately).

I think changing the official name of the current "password" feature is difficult as the term is already rather widespread (for example, it would require fixing dozens of spots in guides), but changing just the option name should pose merely exactly the same problems as the rename of the `server_password` option.

As for the proposed name, I personally don't like the name `room_password` as it doesn't add any specificity, as both password features pertain to "rooms". Further, the concept and terminology of "rooms" really only exists on the WebHost so far. (The one exception perhaps being the two network packets called `RoomInfo` and `RoomUpdate`, but those names are not user-facing.)

As stated before, I don't think changing the option name for the "password" is necessary, but if we are to do that, this certainly would be a good opportunity, and I would propose the name `connect_password`.

**Comment by dewiniaid** [comment]
Just a small datapoint here: In a multiworld today, I did `/option password x` by mistake followed by `/option admin_password x` when I was trying to set the admin password.  So, given that was from (faulty) memory, it suggests that it might be the more intuitive name (and renaming base `password` would prevent me having accidentally set a connect password in the first place.)

**Comment by recklesscoder** [comment]
I feel like this PR will never be merged, and I am tired of resolving conflicts with main, so I'll just close this now.
